src/common/providers/tasks.ts

```diff
@@ -119,7 +169,26 @@ export function onDispatchHandler<Req = any>(
         throw new https.HttpsError("invalid-argument", "Bad Request");
       }
 
-      const context: TaskContext = {};
+      const headers: Record<string, string> = {};
+      for (const [key, value] of Object.entries(req.headers)) {
+        if (!Array.isArray(value)) {
+          headers[key] = value;
+        }
+      }
+
+      const context: TaskContext = {
+        queueName: req.header("X-CloudTasks-QueueName"),
+        id: req.header("X-CloudTasks-TaskName"),
+        retryCount: Number(req.header("X-CloudTasks-TaskRetryCount")),
```

Comment:
nit* Can we make sure this won't below up if the request header isn't missing for some reason (e.g. when we invoke it in the emulator).

Would prefer having safe default behavior in case these headers are missing for whatever reason (maybe log a warning when that happens?)

Reply:
oh i thought this would blow up:

```
> Number(undefined)
NaN
```

Not sure if I like `NaN` but not as bad as I thought.